### PR TITLE
Fix icons in ConfirmTx and TxReceipt

### PR DESCRIPTION
**Pull request by vilsbole**
[[ch8377]](https://app.clubhouse.io/mycrypto/story/8377/send-flow-token-icon-placement-and-shape-are-off)

- Use Currency for amount display with Icon.
- Solve Icon ration in AmountInput.

### CODE REVIEW

File: src/components/Icon/Icon.tsx [hunk 1]
```diff
@@ -360,6 +362,8 @@ const Icon = ({ type, color, ...props }: Props) => {
     return <SDeleteIcon src={svgIcons['nav-close']} color={color} {...props} />;
   } else if (type === 'address-book') {
     return <SInlineSVG src={svgIcons[type]} fill="none" {...props} />;
+  } else if (type === 'tx-wallet') {
+    return <SInlineSVG src={svgIcons[type]} fill="none" {...props} />;
```
Comment (FrederikBolding) [inline src/components/Icon/Icon.tsx:366]: This icon still looks wrong. Also it is the same as `icn-sent`

File: src/components/Icon/Icon.tsx [hunk 2]
```diff
@@ -209,6 +210,7 @@ const svgIcons = {
   'tx-fee': feeIcon,
```
Comment (FrederikBolding) [inline src/components/Icon/Icon.tsx:210]: This icon is wrong too

File: src/components/TokenMigration/components/TokenMigrationForm.tsx [hunk 3]
```diff
@@ -196,22 +196,22 @@ export const TokenMigrationFormUI = ({
...
-              setFieldTouched('amount');
-            }}
-            placeholder={'0.00'}
-          />
-          {errors && errors.amount && touched && touched.amount ? (
-            <InlineMessage className="SendAssetsForm-errors">{errors.amount}</InlineMessage>
-          ) : null}
+          <Label htmlFor="amount">{translate('SEND_ASSETS_AMOUNT_LABEL')}</Label>
         </Tooltip>
+        <AmountInput
```
Comment (FrederikBolding) [inline src/components/TokenMigration/components/TokenMigrationForm.tsx:202]: What is the reasoning behind these changes? It makes more sense imo that the notice shows when hovering the disabled field.

File: src/components/TransactionFlow/ConfirmTransaction.tsx [hunk 4]
```diff
@@ -1,11 +1,9 @@
...
 
-import Styled from 'styled-components';
+import styled from 'styled-components';
 
-import feeIcon from '@assets/images/icn-fee.svg';
-import sendIcon from '@assets/images/icn-send.svg';
-import walletIcon from '@assets/images/icn-wallet.svg';
-import { Amount, AssetIcon, Button, InlineMessage, PoweredByText } from '@components';
+import { Amount, Button, InlineMessage, PoweredByText } from '@components';
+import Icon from '@components/Icon';
```
Comment (FrederikBolding) [inline src/components/TransactionFlow/ConfirmTransaction.tsx:6]: Any reason to not directly import?

File: src/components/TransactionFlow/ConfirmTransaction.tsx [hunk 5]
```diff
@@ -272,79 +255,84 @@ export const ConfirmTransactionUI = ({
...
               amount: convertToFiat(bigify(amount), assetRate).toFixed(2)
             }}
           />
-        </AmountWrapper>
+        </ColumnWrapper>
       </RowWrapper>
       <RowWrapper>
         <ColumnWrapper>
-          <img src={feeIcon} alt="Fee" /> {translate('CONFIRM_TX_FEE')}
+          <SIcon type="tx-fee" alt="Fee" height="25px" />
```
Comment (FrederikBolding) [inline src/components/TransactionFlow/ConfirmTransaction.tsx:280]: I would argue that using `img` for SVGs that we don't need to modify might be less of a PITA than what we are currently doing. It shouldn't be this difficult to use an image

File: src/components/TransactionFlow/TxReceiptTotals.tsx [hunk 6]
```diff
@@ -2,14 +2,23 @@ import React from 'react';
...
 import { ExtendedAsset, ISettings } from '@types';
 import { bigify, convertToFiat, fromWei, totalTxFeeToWei, Wei } from '@utils';
 
+const SIcon = styled(Icon)`
+  width: 30px;
+  height: 30px;
+  margin-right: ${SPACING.SM};
+  vertical-align: middle;
+  @media (max-width: ${BREAK_POINTS.SCREEN_SM}) {
+    display: none;
```
Comment (FrederikBolding) [inline src/components/TransactionFlow/TxReceiptTotals.tsx:19]: What is the reasoning behind hiding them on mobile?

**Review by FrederikBolding — CHANGES_REQUESTED** [review]

File: src/components/Amount.tsx [hunk 7]
```diff
@@ -1,85 +1,80 @@
...
+    uuid?: TUuid;
+  };
   fiat?: {
+    // Converted Fiat
     amount: string;
     symbol: TCurrencySymbol;
     ticker: TTicker;
   };
-  baseAssetValue?: string;
+  baseAssetValue?: string; // ?
```
[the remark this answers is not in the document]
Comment (vilsbole) [reply src/components/Amount.tsx:24]: Updated comment.

File: src/components/Amount.tsx [hunk 8]
```diff
@@ -1,85 +1,80 @@
...
-  @media (min-width: ${BREAK_POINTS.SCREEN_XS}) {
-    font-size: 1em;
-  }
-`;
+import { Text } from './NewTypography';
 
 interface Props {
-  assetValue: string;
+  assetValue?: string; // Free string display
+  asset?: {
```
[the remark this answers is not in the document]
Comment (vilsbole) [reply src/components/Amount.tsx:12]: Will be addressed with https://app.clubhouse.io/mycrypto/story/8439/harmonise-amount-usage

File: src/components/Amount.tsx [hunk 9]
```diff
@@ -1,85 +1,80 @@
...
       {fiat && (
-        <SCurrency
+        <Currency
           amount={fiat.amount}
           symbol={fiat.symbol}
           ticker={fiat.ticker}
           decimals={2}
-          fiatColor={fiatColor}
+          color={fiatColor}
+          fontSize="0.9em"
```
[the remark this answers is not in the document]
Comment (vilsbole) [reply src/components/Amount.tsx:75]: Yes per design: https://mycryptohq.slack.com/files/UQR1QG77A/F01SU0ZCG3C/screen_shot_2021-03-30_at_9.38.54_am.png

File: src/components/TransactionFlow/ConfirmTransaction.tsx (hunk 4 above)
Comment (vilsbole) [reply src/components/TransactionFlow/ConfirmTransaction.tsx:6]: The imports break when I do.
Most likely due to circular dependencies. eg. A component importing another from the same `index.ts`

File: src/components/TransactionFlow/TxReceiptTotals.tsx (hunk 6 above)
Comment (vilsbole) [reply src/components/TransactionFlow/TxReceiptTotals.tsx:19]: Design hides the asset icon on mobile. https://www.figma.com/file/BY0SWc75teEUZzws8JdgLMpy/MyCrypto-Web-App-Final-Designs?node-id=701%3A1523. Most likely due to save real estate and avoid wrapping. 

I find that the asset icon is more important than the title icon, so I inverted the rule and hide the latter.

File: src/components/Icon/Icon.tsx (hunk 1 above)
Comment (vilsbole) [reply src/components/Icon/Icon.tsx:366]: Corrected to use `tx-sent` aka `icn-sent`

File: src/components/TransactionFlow/ConfirmTransaction.tsx (hunk 5 above)
Comment (vilsbole) [reply src/components/TransactionFlow/ConfirmTransaction.tsx:280]: No objection to exporting img tag from Icon in the future.

File: src/components/Icon/Icon.tsx (hunk 2 above)
Comment (vilsbole) [reply src/components/Icon/Icon.tsx:210]: Fixed

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**